### DISCUSSION TITLE
When to use `errors.WithStack(err)` in the code?

### DISCUSSION THREAD

**Discussion by carlisia** [vmware-tanzu/velero#3426] [Community support Q&A] [answered]
In the code guidelines here: https://velero.io/docs/v1.5/code-standards/#code it says:

> Wrap/add a stack only to errors that are being directly returned from non-velero code, such as an API call to the Kubernetes server.

Can this be clarified?

For example, I would think the error below would qualify to be logged with stack, but as we can see from the output of the log message when using with/without stack it is the same:

https://github.com/vmware-tanzu/velero/blob/3286834a906de2bbd4109400a6b558973010b9ab/pkg/controller/backup_sync_controller.go#L182

![image](https://user-images.githubusercontent.com/16508/107268395-eb94ba80-69fc-11eb-820b-d98d12a22e96.png)

![image](https://user-images.githubusercontent.com/16508/107268409-efc0d800-69fc-11eb-8083-cfa4128eb1f2.png)

**Comment by nrb** [comment, marked as answer]
I interpret the documentation to mean "_Any_ Go package that is not within the Velero codebase, including standard library, should be returned with errors.WithStack."

Other functions like `errrors.Wrap` actually do `WithStack` under the hood, if I remember correctly.
